be/src/exec/pipeline/olap_chunk_source.cpp

```diff
@@ -173,12 +188,7 @@ Status OlapChunkSource::_init_reader_params(const std::vector<OlapScanRange*>& k
     _params.profile = _runtime_profile;
     _params.runtime_state = _runtime_state;
     _params.use_page_cache = !config::disable_storage_page_cache;
-    // Improve for select * from table limit x, x is small
-    if (_limit != -1 && _limit < _runtime_state->chunk_size()) {
-        _params.chunk_size = _limit;
-    } else {
-        _params.chunk_size = _runtime_state->chunk_size();
-    }
+    _decide_chunk_size();
```

Comment:
Why not do this in FE when planning?

Reply:
`chunk_size` is a global parameter, which affects all operators.
Current bad case is located at scan operator, the IO thread allocates chunk and transfer to computation thread, which cause memory allocation bottleneck.

Current solution is just a work-around, we need a better memory allocator to solve there issues.

Reply:
Oh I see.
But I have another question, if some other ExecNodes will generate `huge` type, will it have the memory allocation problem and affect the performance?
IMO, we can give a smaller(like 4096) global chunk_size as long as there is a `huge` type in the whole query.

Reply:
In theory it has.
And the whole solution may be, each operator needs to decide the chunk size individually, according to its input type and output type.

For this solution, we need to benchmark carefully, which is not easy work. 

And currently we need to focus at the performance degradation between pipeline & non-pipeline mode, and solve this issue at `version-2.2`, and leave the adaptive chunk_size & query-intrinsic memory allocator to future work.